src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidTagsCheck.java

```diff
@@ -37,21 +52,87 @@
             "OSM feature {0,number,#} has invalid tags.",
             "Check the following tags for missing, conflicting, or incorrect values: {0}");
     private static final String KEY_VALUE_SEPARATOR = "->";
+    private static final String DEFAULT_FILTER_RESOURCE = "invalidTags.txt";
+    private static final Logger logger = LoggerFactory.getLogger(InvalidTagsCheck.class);
+
+    private final List<Tuple<? extends Class<AtlasEntity>, List<TaggableFilter>>> classTagFilters;
+    private final boolean overrideDefaultFilters;
+    private final boolean appendToDefaultFilters;
 
-    private final List<Tuple<? extends Set<Class<AtlasEntity>>, TaggableFilter>> classTagFilters;
+    /**
+     * @return a List of Tuple containing AtlasEntity and a list of TaggableFilters read from the
+     *         json files of each AtlasEntity. DEFAULT_FILTER_RESOURCE file maps each AtlasEntity to
+     *         its corresponding filter files.
+     */
+    private static List<Tuple<? extends Class<AtlasEntity>, List<TaggableFilter>>> getDefaultFilters()
+    {
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(
+                InvalidTagsCheck.class.getResourceAsStream(DEFAULT_FILTER_RESOURCE))))
+        {
+            final List<Tuple<? extends Class<AtlasEntity>, List<TaggableFilter>>> listOfClassToFilters = new ArrayList<>();
+            String line;
+            while ((line = reader.readLine()) != null)
+            {
+                final String[] split = line.split(DELIMITER);
+                if (split.length == 2)
+                {
+                    listOfClassToFilters.add(
+                            new Tuple<>(ItemType.valueOf(split[0].toUpperCase()).getMemberClass(),
+                                    getFiltersFromResource(split[1])));
+                }
+            }
+            return listOfClassToFilters;
+        }
+        catch (final IOException exception)
+        {
+            logger.error(String.format("Could not read %s", DEFAULT_FILTER_RESOURCE), exception);
+            return Collections.emptyList();
+        }
+    }
 
     /**
      * Gathers the keys from a {@link TaggableFilter} using regex.
      *
-     * @param filter
+     * @param listOfFilters<TaggableFilter>
      *            a {@link TaggableFilter}
      * @return the tag keys as a {@link Set} of {@link String}s
      */
-    private static Set<String> getFilterKeys(final TaggableFilter filter)
+    private static Set<String> getFilterKeys(final AtlasObject atlasObject,
+            final List<TaggableFilter> listOfFilters)
     {
-        return Arrays.stream(filter.toString().split("[|&]+"))
-                .filter(string -> string.contains(KEY_VALUE_SEPARATOR))
-                .map(tag -> tag.split(KEY_VALUE_SEPARATOR)[0]).collect(Collectors.toSet());
+        final Set<String> collectedKeys = new HashSet<>();
+        listOfFilters.stream().filter(taggableFilter -> taggableFilter.test(atlasObject)).forEach(
```

Comment:
This method was originally made to return the keys from a single filter so that an instruction could be made for each filter that is passed. Now this can return the combined keys of multiple filters, which could make the instructions more confusing.

Reply:
Hmm yeah. I am filtering the list of filters here so that only the keys of the filter which the atlasobject passes will be used in the instruction. But I think it makes sense to group the keys in the instruction to avoid confusion.

Reply:
@Bentleysb I have updated the logic to pass taggable filter, instead of list of filters, to this method so that for each filter, a new instruction is created just as before.